new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainOnPV.java

```diff
@@ -0,0 +1,706 @@
+    List<Path> domainScriptFiles = new ArrayList<>();
+    domainScriptFiles.add(targetWlstScript);
+    domainScriptFiles.add(domainPropertiesFile);
+
+    logger.info("Create a config map to hold domain creation scripts");
+    String domainScriptConfigMapName = "create-domain-scripts-cm";
+    assertDoesNotThrow(
+        () -> createConfigMapForDomainCreation(domainScriptConfigMapName, domainScriptFiles),
+        "Creating configmap for domain creation failed");
+
+    logger.info("Running a Kubernetes job to create the domain");
+    runCreateDomainJob(pvName, pvcName, domainScriptConfigMapName, domainNamespace);
+
+  }
+
+  /**
+   * Create a properties file for WebLogic domain configuration.
+   * @param wlstPropertiesFile path of the properties file
+   * @throws FileNotFoundException when properties file path not found
+   * @throws IOException when writing properties fails
+   */
+  private void createDomainProperties(Path wlstPropertiesFile) throws FileNotFoundException, IOException {
+    // create a list of properties for the WebLogic domain configuration
+    Properties p = new Properties();
+
+    p.setProperty("domain_path", "/shared/domains");
+    p.setProperty("domain_name", domainUid);
+    p.setProperty("cluster_name", clusterName);
+    p.setProperty("admin_server_name", adminServerName);
+    p.setProperty("managed_server_port", "8001");
+    p.setProperty("admin_server_port", "7001");
+    p.setProperty("admin_username", "system");
+    p.setProperty("admin_password", "gumby1234");
+    p.setProperty("admin_t3_public_address", K8S_NODEPORT_HOST);
+    p.setProperty("admin_t3_channel_port", "32001");
+    p.setProperty("number_of_ms", "4");
+    p.setProperty("managed_server_name_base", managedServerNameBase);
+    p.setProperty("domain_logs", "/shared/logs");
+    p.setProperty("production_mode_enabled", "true");
+
+    p.store(new FileOutputStream(wlstPropertiesFile.toFile()), "wlst properties file");
+  }
+
+  /**
+   * Create configmap containing domain scripts.
+   * @param configMapName name of the configmap to create
+   * @param files files to add in configmap
+   * @throws IOException when reading the file fails
+   * @throws ApiException when create configmap fails
+   */
+  private void createConfigMapForDomainCreation(String configMapName, List<Path> files)
+      throws IOException, ApiException {
+
+    // add wlst domain creation python script and properties files
+    // to create domain to the configmap
+    Map<String, String> data = new HashMap<>();
+    for (Path file : files) {
+      data.put(file.getFileName().toString(), Files.readString(file));
+    }
+    V1ObjectMeta meta = new V1ObjectMeta()
+        .name(configMapName)
+        .namespace(domainNamespace);
+    V1ConfigMap configMap = new V1ConfigMap()
+        .data(data)
+        .metadata(meta);
+
+    boolean cmCreated = assertDoesNotThrow(() -> TestActions.createConfigMap(configMap),
```

Comment:
You can static import TestActions.createConfigMap

Reply:
fixed